src/main/java/com/endava/endabank/service/impl/UserAuthenticationService.java

```diff
@@ -0,0 +1,58 @@
+package com.endava.endabank.service.impl;
+
+import com.endava.endabank.constants.Strings;
+import com.endava.endabank.dao.UserDao;
+import com.endava.endabank.exceptions.customexceptions.BadDataException;
+import com.endava.endabank.model.Role;
+import com.endava.endabank.model.User;
+import com.endava.endabank.security.UserAuthentication;
+import com.endava.endabank.security.utils.JwtManage;
+import lombok.AllArgsConstructor;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+
+@Service
+@AllArgsConstructor
+public class UserAuthenticationService implements UserDetailsService {
+    private UserDao userDao;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userDao.findByEmail(username).
+                orElseThrow(() -> new UsernameNotFoundException(Strings.USER_NOT_FOUND));
+        Role role = user.getRole();
+        ArrayList<SimpleGrantedAuthority> authorities = new ArrayList<>();
```

Comment:
Hi,

You could replace these line 35,36 with 
`List<SimpleGrantedAuthority> authorities = List.of(new SimpleGrantedAuthority(role.getName()));`